src/WorkItems/Logging/LoggerExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.WorkItems.Logging
+{
+    public static class LoggerExtensions
+    {
+        public static void LogMetrics(this ILogger logger, EventId eventId, IDictionary<string, object> customDimensions)
```

Comment:
>LogMetrics [](start = 27, length = 10)

Why don't we have a single extension method here that puts the optional message arg at the end? Are you trying to match existing overload patterns on ILogger?

Reply:
Yes. it loosely follows the existing pattern. If we add the message at the end, I think it'll be a bit confusing because it'll be out-of-order compared to the existing ILogger extensions.

---
In reply to: [396834358](https://github.com/microsoft/sarif-sdk/pull/1820#discussion_r396834358) [](ancestors = 396834358)